src/app/components/Account/RuntimeAccountDetailsView.tsx

```diff
@@ -77,7 +77,7 @@ export const RuntimeAccountDetailsView: FC<RuntimeAccountDetailsViewProps> = ({
       </StyledListTitleWithAvatar>
       <dd>
         <AccountLink
-          showAddressAsName
+          showOnlyAddress={!!token?.name}
```

Comment:
Isn't `!!contract` more appropriate here?

Reply:
What is the rationale for hiding the name for all contracts?

(Please note that we can also have other named contracts, not just token contracts. Tokens contracts are already handled.)

Reply:
ah we show other row with name only for `token` so I think we should use the same condition in both places.

Reply:
The name field is mandatory for `ERC-20` tokens, but not for other tokens.
That is why the `name`  field (of the token data structure) is defined like this:
```
  /** Name of the token, as provided by token contract's `name()` method. */
  name?: string;
```

So it's possible that we have tokens without names. So displaying a token link doesn't necessarily mean that we have a valid name there. That's why I used `token.name` for the condition.

This way, if the token doesn't have a name, but the smart contract does, we can still display it.